File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.js
function solution(players, m, k) {
    let result = 0;
    let servers = Array(24).fill(0)
    
    players.forEach((player, idx) => {
        if(parseInt(player / m) > servers[idx]){
            let addServerCnt = parseInt(player / m) - servers[idx]
            
            for(let i = 0 ; i < k ; ++i){
                if(idx + 1 <= 23) servers[idx+i] = servers[idx+i] + addServerCnt
            }
            result += addServerCnt;
        }
    })
    return result
}

File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.py
def solution(players, m, k):
    result = 0
    servers = [0 for _ in range(24)]
    
    for idx, player in enumerate(players):
        if player // m > servers[idx]:
            add_server_cnt = player // m - servers[idx]
            
            for i in range(k):
                if idx + i < 24:
	                servers[idx+i] += add_server_cnt
            
            result += add_server_cnt
    
    return result